pkg/operator/ceph/object/spec.go

```diff
@@ -404,20 +408,43 @@ func (c *clusterConfig) reconcileService(cephObjectStore *cephv1.CephObjectStore
 }
 
 func (c *clusterConfig) vaultPrefixRGW() string {
-	secretEngine := c.clusterSpec.Security.KeyManagementService.ConnectionDetails[kms.VaultSecretEngineKey]
+	secretEngine := c.store.Spec.Security.KeyManagementService.ConnectionDetails[kms.VaultSecretEngineKey]
 	vaultPrefixPath := "/v1/"
 
 	switch secretEngine {
 	case kms.VaultKVSecretEngineKey:
 		vaultPrefixPath = path.Join(vaultPrefixPath,
-			c.clusterSpec.Security.KeyManagementService.ConnectionDetails[vault.VaultBackendPathKey])
+			c.store.Spec.Security.KeyManagementService.ConnectionDetails[vault.VaultBackendPathKey], "/data")
 	case kms.VaultTransitSecretEngineKey:
 		vaultPrefixPath = path.Join(vaultPrefixPath, secretEngine, "/export/encryption-key")
 	}
 
 	return vaultPrefixPath
 }
 
+func (c *clusterConfig) CheckRGWKMS() (bool, error) {
```

Comment:
We really need unit test for this.

Reply:
@leseb : please refer spec_test.go TestCheckRGWKMS()